mirdata/datasets/queen.py

```diff
@@ -0,0 +1,267 @@
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.chords_path = self.get_path("chords")
+        self.keys_path = self.get_path("keys")
+        self.sections_path = self.get_path("sections")
+        self.audio_path = self.get_path("audio")
+
+        self.title = os.path.basename(self.get_path("sections")).split(".")[0]
+
+    @core.cached_property
+    def chords(self) -> Optional[annotations.ChordData]:
+        return load_chords(self.chords_path)
+
+    @core.cached_property
+    def key(self) -> Optional[annotations.KeyData]:
+        return load_key(self.keys_path)
+
+    @core.cached_property
+    def sections(self) -> Optional[annotations.SectionData]:
+        return load_sections(self.sections_path)
+
+    @property
+    def audio(self) -> Tuple[np.ndarray, float]:
+        """The track's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """the track's data in jams format
+
+        Returns:
+            jams.JAMS: return track data in jam format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            section_data=[(self.sections, None)],
+            chord_data=[(self.chords, None)],
+            key_data=[(self.key, None)],
+            metadata={"artist": "Queen", "title": self.title},
+        )
+
+
+def load_audio(fhandle: str) -> Tuple[np.ndarray, float]:
+    """Load a TONAS audio file.
+
+    Args:
+        fhandle (str): path to an audio file
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+
+    """
+    return librosa.load(fhandle, sr=44100, mono=True)
+
+
+@io.coerce_to_string_io
+def load_chords(fhandle: TextIO) -> annotations.ChordData:
+    """Load Queen format chord data from a file
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to a chord file
+
+    Returns:
+        (ChordData): loaded chord data
+
+    """
+    start_times, end_times, chords = [], [], []
+    f = open(fhandle.name, "r")
+    reader = csv.reader(f, delimiter="\t")
+    for line in reader:
+        start_times.append(float(line[0]))
+        end_times.append(float(line[1]))
+        chords.append(line[2])
+
+    return annotations.ChordData(np.array([start_times, end_times]).T, chords)
+
+
+@io.coerce_to_string_io
+def load_key(fhandle: TextIO) -> annotations.KeyData:
+    """Load Queen format key data from a file
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to a key file
+
+    Returns:
+        (KeyData): loaded key data
+
+    """
+    start_times, end_times, keys = [], [], []
+    f = open(fhandle.name, "r")
```

Comment:
same comment here - fhandle is already an open file!

Reply:
done

Reply:
done!